Not fully compatible with netstandard20

Hi,
I'm referencing the package from a netstandard2.0 project, and VS displays this warning:

> warning NU1701: Package 'SharedMemory 2.1.0' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.

Is this expected?  Are there any risks in ignoring this warning?

Edit: I guess the library projects should target netstandard2.0 instead of netcoreapp2.0.

Brgds,
Robert

--- a/.gitignore
+++ b/.gitignore
@@ -21,3 +21,5 @@ obj/
 _ReSharper*/
 TestResults*/
 *.nupkg
+/.vs/SharedMemory
+/nuget.exe

--- a/Examples/ClientTest/ClientTest.csproj
+++ b/Examples/ClientTest/ClientTest.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp2.0;net45;net4;net35</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0;net47;net46;net45;net4;net35</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>

--- a/Examples/ServerTest/ServerTest.csproj
+++ b/Examples/ServerTest/ServerTest.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp2.0;net45;net4;net35</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0;net47;net46;net45;net4;net35</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>

--- a/Examples/SingleProcess/SingleProcess.csproj
+++ b/Examples/SingleProcess/SingleProcess.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp2.0;net45;net4;net35</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0;net47;net46;net45;net4;net35</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>

--- a/SharedMemory.Tests/SharedMemory.Tests.csproj
+++ b/SharedMemory.Tests/SharedMemory.Tests.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.0;net45</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0;net47;net46;net45</TargetFrameworks>
 
     <IsPackable>false</IsPackable>
   </PropertyGroup>

--- a/SharedMemory.nuspec
+++ b/SharedMemory.nuspec
@@ -21,6 +21,10 @@ It features:
 Usage: http://sharedmemory.codeplex.com/documentation</description>
         <summary>Shared memory classes for sharing data between processes (Array, Buffer and Circular Buffer)</summary>
         <releaseNotes>$version$
+1. Changed from .NET Core 2 target to .NET Standard 2.0 - this requires a new NuGet dependency to support DynamicMethod
+1. Added 4.6/4.7 targets
+
+2.1.0
 1. Add .NET Core 2 target and update to use dotnet project files (VS2017) (#29)
         
 2.0.16
@@ -53,10 +57,20 @@ Usage: http://sharedmemory.codeplex.com/documentation</description>
           <group targetFramework=".NETFramework3.5" />
           <group targetFramework=".NETFramework4.0" />
           <group targetFramework=".NETFramework4.5" />
-          <group targetFramework=".NETCoreApp2.0" />
+          <group targetFramework=".NETFramework4.6" />
+          <group targetFramework=".NETFramework4.7" />
+          <group targetFramework=".NETStandard2.0">
+              <dependency id="System.Reflection.Emit.Lightweight" version="4.3.0" />
+          </group>
         </dependencies>     
     </metadata>
     <files>
+        <file src=".\bin\Release\Net45\SharedMemory.dll" target="lib\net47" />
+        <file src=".\bin\Release\Net45\SharedMemory.pdb" target="lib\net47" />
+        <file src=".\bin\Release\Net45\SharedMemory.xml" target="lib\net47" />
+        <file src=".\bin\Release\Net45\SharedMemory.dll" target="lib\net46" />
+        <file src=".\bin\Release\Net45\SharedMemory.pdb" target="lib\net46" />
+        <file src=".\bin\Release\Net45\SharedMemory.xml" target="lib\net46" />	
         <file src=".\bin\Release\Net45\SharedMemory.dll" target="lib\net45" />
         <file src=".\bin\Release\Net45\SharedMemory.pdb" target="lib\net45" />
         <file src=".\bin\Release\Net45\SharedMemory.xml" target="lib\net45" />
@@ -66,8 +80,9 @@ Usage: http://sharedmemory.codeplex.com/documentation</description>
         <file src=".\bin\Release\Net35\SharedMemory.dll" target="lib\net35" />
         <file src=".\bin\Release\Net35\SharedMemory.pdb" target="lib\net35" />
         <file src=".\bin\Release\Net35\SharedMemory.xml" target="lib\net35" />
-        <file src=".\bin\Release\netcoreapp2.0\SharedMemory.dll" target="lib\netcoreapp2.0" />
-        <file src=".\bin\Release\netcoreapp2.0\SharedMemory.xml" target="lib\netcoreapp2.0" />
-        <file src=".\SharedMemory\**\*.cs" target="src" />
+        <file src=".\bin\Release\netstandard2.0\SharedMemory.dll" target="lib\netstandard2.0" />
+        <file src=".\bin\Release\netstandard2.0\SharedMemory.pdb" target="lib\netstandard2.0" />
+        <file src=".\bin\Release\netstandard2.0\SharedMemory.xml" target="lib\netstandard2.0" />
+        <file src=".\SharedMemory\**\*.cs" exclude=".\SharedMemory\obj\**\*.*" target="src" />
     </files>
 </package>

--- a/SharedMemory/SharedMemory.csproj
+++ b/SharedMemory/SharedMemory.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.0;net45;net4;net35</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net47;net46;net45;net4;net35</TargetFrameworks>
     <OutputPath>..\bin\$(Configuration)\</OutputPath>
     <DocumentationFile>..\bin\$(Configuration)\$(TargetFramework)\SharedMemory.xml</DocumentationFile>
   </PropertyGroup>
@@ -14,10 +14,18 @@
     <Copyright>Copyright (c) 2018 Justin Stenning</Copyright>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.0'">
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0'">
     <DefineConstants>NETCORE;NETCORE2_0;NET40Plus</DefineConstants>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net47'">
+    <DefineConstants>NET45;NETFULL;NET40Plus</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net46'">
+    <DefineConstants>NET45;NETFULL;NET40Plus</DefineConstants>
+  </PropertyGroup>
+  
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net45'">
     <DefineConstants>NET45;NETFULL;NET40Plus</DefineConstants>
   </PropertyGroup>
@@ -29,5 +37,11 @@
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net35'">
     <DefineConstants>NET35;NETFULL</DefineConstants>
   </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="System.Reflection.Emit.Lightweight">
+      <Version>4.3.0</Version>
+    </PackageReference>
+  </ItemGroup>
 
 </Project>

